main/src/cgeo/geocaching/HandleLocalFilesActivity.java

```diff
@@ -29,10 +34,10 @@ public void onCreate(final Bundle savedInstanceState) {
                 final int startExtension = parsedUri.lastIndexOf('.');
                 if (startExtension > -1) {
                     final String extension = parsedUri.substring(startExtension).toLowerCase();
-                    if (".zip".equals(extension) || ".gpx".equals(extension) || ".gpx".equals(extension)) {
+                    if (ZIP_FILE_EXTENSION.equalsIgnoreCase(extension) || GPX_FILE_EXTENSION.equalsIgnoreCase(extension) || COMPRESSED_GPX_FILE_EXTENSION.equalsIgnoreCase(extension) || LOC_FILE_EXTENSION.equalsIgnoreCase(extension)) {
```

Comment:
```suggestion
                    if (extension.equals(ZIP_FILE_EXTENSION) || extension.equals(GPX_FILE_EXTENSION) || extension.equals(COMPRESSED_GPX_FILE_EXTENSION) || extension.equals(LOC_FILE_EXTENSION)) {
```

Reply:
In know this looks a bit awkward, but the other way around Codacy complains with "Issue found: Position literals first in String comparisons"

Reply:
But we are using constants now, not literals.

Reply:
Ah, I see. Thanks for pointing that out to me. - I thought, Codacy would handle them the same. Have updated the PR now.